op-devstack/sysgo/l2_metrics_dashboard.go

```diff
@@ -0,0 +1,325 @@
+package sysgo
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/op-devstack/devtest"
+	"github.com/ethereum-optimism/optimism/op-devstack/stack"
+	"github.com/ethereum-optimism/optimism/op-service/locks"
+	"github.com/ethereum-optimism/optimism/op-service/logpipe"
+	"gopkg.in/yaml.v3"
+)
+
+const DockerExecutablePathEnvVar = "SYSGO_DOCKER_EXEC_PATH"
+const GrafanaProvisioningDirEnvVar = "SYSGO_GRAFANA_PROVISIONING_DIR"
+const GrafanaDataDirEnvVar = "SYSGO_GRAFANA_DATA_DIR"
+
+const DockerToLocalHost = "host.docker.internal"
+
+const PrometheusHost = "0.0.0.0"
+const PrometheusServerPort = "9999"
+const PrometheusDockerPort = "9090"
+
+const GrafanaHost = "0.0.0.0"
+const GrafanaServerPort = "3000"
+const GrafanaDockerPort = "3000"
+
+type L2MetricsDashboard struct {
+	p devtest.P
+
+	grafanaExecPath   string
+	grafanaArgs       []string
+	grafanaEnv        []string
+	grafanaSubprocess *SubProcess
+
+	prometheusExecPath   string
+	prometheusArgs       []string
+	prometheusEnv        []string
+	prometheusSubprocess *SubProcess
+
+	prometheusEndpoint string
+}
+
+func (g *L2MetricsDashboard) Start() {
+	g.startPrometheus()
+	g.startGrafana()
+}
+
+func (g *L2MetricsDashboard) Stop() {
+	err := g.grafanaSubprocess.Stop()
+	g.p.Require().NoError(err, "Grafana must stop")
+
+	err = g.prometheusSubprocess.Stop()
+	g.p.Require().NoError(err, "Prometheus must stop")
+
+	g.grafanaSubprocess = nil
+	g.prometheusSubprocess = nil
+}
+
+func (g *L2MetricsDashboard) startPrometheus() {
+	// Create the sub-process.
+	// We pipe sub-process logs to the test-logger.
+	logOut := logpipe.ToLogger(g.p.Logger().New("src", "stdout"))
+	logErr := logpipe.ToLogger(g.p.Logger().New("src", "stderr"))
```

Comment:
Wondering if we want to add more logging context so that it's clear this is coming from prometheus, like

```suggestion
	logOut := logpipe.ToLogger(g.p.Logger().New("component", "prometheus", "src", "stdout"))
	logErr := logpipe.ToLogger(g.p.Logger().New("component", "prometheus", "src", "stderr"))
```

Reply:
https://github.com/ethereum-optimism/optimism/pull/18017/commits/2b5bfea54d5871e5c80dca0c3751dcf454f55c2f